### ISSUE TITLE
[Bug]: [FieldCollection] Data Loss if Relation has been deleted

### ISSUE THREAD

**Issue by solverat** [pimcore/pimcore#15448]
### Pimcore version

10.5.x

### Steps to reproduce

If you're removing relation(s) from a Fieldcollection via PHP API, all relations will be removed after resaving the object.

Steps to reproduce:
```php

$c = MyObject::getById(100);

$fc = new Fieldcollection();

$asset1 = new Asset();
$asset1->setParent(Asset::getByPath('/'));
$asset1->setKey('test-1');
$asset1->save();

$asset2 = new Asset();
$asset2->setParent(Asset::getByPath('/test'));
$asset2->setKey('test-2');
$asset2->save();

$fcb1 = new Fieldcollection\Data\DocumentsBlock();
$fcb1->setIdentifier('test1');
$fcb1->setDocuments([new ElementMetadata('documents', [], $asset1)]);

$fcb2 = new Fieldcollection\Data\DocumentsBlock();
$fcb2->setIdentifier('test2');
$fcb2->setDocuments([new ElementMetadata('documents', [], $asset2)]);

$fc->add($fcb1);
$fc->add($fcb2);

$c->setDocuments($fc);
$c->save(); // result: see Screenshot I

// now delete a relation
$asset2->delete();

$freshC = MyObject::getById(100, ['force' => true]);
$freshC->save();  // result: see Screenshot II
```

## Screenshot I
![image](https://github.com/pimcore/pimcore/assets/700119/ccd13c42-6a2f-4d2f-aa1d-bc2986976610)

## Screenshot II
![image](https://github.com/pimcore/pimcore/assets/700119/c3a579b7-42d7-495f-b2da-9ae5eb1622aa)

BTW: Didn't check it on 11, but i guess it's the same there.

### Expected / actual behavior

Only the relation gets removed, other fields should not be touched.

**Comment by solverat** [comment]
Btw. Adding `DataObject::disableDirtyDetection(true);` fixes this issue.

**Comment by kingjia90** [comment]
Nevermind, could reproduce it

~Haven't reproduced it yet, but i've noticed something, could it be some kind of allowed element sanitization?`~

~In the code, it is assigning an new `Asset` (`$asset1 = new Asset();`) to something restricted to `Documents`?~
`$fcb1->setDocuments([new ElementMetadata('documents', [], $asset1)]);`

> Btw. Adding `DataObject::disableDirtyDetection(true);` fixes this issue.

May be also linked, when dirty detection is disabled, it's not perfoming any sanitization? ~

**Comment by kingjia90** [comment]
was able to reproduce it, sharing some finding must be some race condition or so, it works if i put a `sleep(3);` 😰 
https://github.com/pimcore/pimcore/blob/e469e8641a6b6bfbcd4e38c7dd03b226f6081a52/models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php#L112-L113




and also found that this `$data` is not used anywhere, not sure if it's relevant
https://github.com/pimcore/pimcore/blob/e469e8641a6b6bfbcd4e38c7dd03b226f6081a52/models/DataObject/ClassDefinition/Data/Relations/ManyToManyRelationTrait.php#L46

**Comment by solverat** [comment]
The reason I THINK is, that **all** fieldcollection relations will be removed if something has changed, but **only that specific** fieldcollection property receives the dirty flag (which will be removed in my example).

**Comment by kingjia90** [comment]
Seems something like that 
https://github.com/pimcore/pimcore/blob/8dfdbce7be25ffd96142d5dd268cf649b2d965aa/models/DataObject/Fieldcollection/Dao.php#L258
(Despite function name is delete, it has a Savemode parameter)
while returns differently on 
https://github.com/pimcore/pimcore/blob/8dfdbce7be25ffd96142d5dd268cf649b2d965aa/models/DataObject/Fieldcollection/Dao.php#L250 when disabling dirty detection or so

**Comment by kingjia90** [comment]
By looking at the history/version, i've noticed the sanitization is removing that.
![image](https://github.com/pimcore/pimcore/assets/6014195/2f215c43-9182-4e7d-a404-b588e4896b06)
![image](https://github.com/pimcore/pimcore/assets/6014195/584effa9-de2c-4274-8c34-4815fbc24bd9)
So checking on this track

probably racing condition, the sanitization uses the first object with Wasd as 123 
![image](https://github.com/pimcore/pimcore/assets/6014195/cd3c795b-208b-4143-8f7d-e39adffbd56d)
![image](https://github.com/pimcore/pimcore/assets/6014195/a4c8772b-2eac-4ec7-833c-74be902bcaef)

see also https://github.com/pimcore/pimcore/blob/e469e8641a6b6bfbcd4e38c7dd03b226f6081a52/lib/Messenger/Handler/SanityCheckHandler.php#L49

note also another bug, the "first" version with123 should have all 3 relations saved, instead it displays as if is after $asset2 deletion

**Comment by kingjia90** [comment]
Ok, confirmed the fix, backed by tests https://github.com/pimcore/pimcore/pull/15483/commits/45b16857811ee94aee5b9c507dd5390d61271eb7 as well

**Comment by dvesh3** [comment]
Fixed by #15483

**Comment by solverat** [comment]
@kingjia90 Thanks for your efforts! 💪
